examples/tls-sni/envoy-client.yaml

```diff
@@ -0,0 +1,95 @@
+static_resources:
+  listeners:
+  - address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 20000
+    listener_filters:
+    - name: "envoy.filters.listener.tls_inspector"
```

Comment:
im unclear about this

in the sni section "securing envoy" page i did not mention it, so i  tested out here whether it was needed to make the sni work and it doesnt seem to be

reading the  docs again (https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/tls_inspector#tls-inspector) it seems like it does 2 things - make sni work, and add stats

it doesnt seem like it really is required to make sni work - it this correct ?